appinventor/components/src/com/google/appinventor/components/runtime/FirebaseDB.java

```diff
@@ -0,0 +1,347 @@
+import com.google.appinventor.components.annotations.SimpleFunction;
+import com.google.appinventor.components.annotations.SimpleObject;
+import com.google.appinventor.components.annotations.SimpleProperty;
+import com.google.appinventor.components.annotations.UsesLibraries;
+import com.google.appinventor.components.annotations.UsesPermissions;
+import com.google.appinventor.components.common.ComponentCategory;
+import com.google.appinventor.components.common.PropertyTypeConstants;
+import com.google.appinventor.components.common.YaVersion;
+
+import android.app.Activity;
+import android.os.Handler;
+
+// When the component is installed in App Inventor, the Javadoc
+// comments will become included in the automatically-generated system
+// documentation, except for lines starting with tags (such as @author).
+/**
+ * The Firebase component communicates with a Web service to store
+ * and retrieve information.  The component has methods to
+ * store a value under a tag and to retrieve the value associated with
+ * the tag. It also possesses a listener to fire events when stored 
+ * values are changed.
+ *
+ * @author kasmus@mit.edu (Kristin Asmus)
+ */
+
+
+// The annotations here provide information to the compiler about
+// integrating the component into App Inventor system.  The following
+// three annotations stipulate that Firebase Component will appear in the
+// designer, that it will be an object in the App Inventor language,
+// and say what Android system permissions it requires.
+
+@DesignerComponent(version = YaVersion.FIREBASE_COMPONENT_VERSION,
+    description = "Non-visible component that communicates with Firebase to store and " +
+    "retrieve information.",
+    category = ComponentCategory.STORAGE,
+    nonVisible = true,
+    iconName = "images/tinyWebDB.png") //FIXME
+@SimpleObject
+@UsesPermissions(permissionNames = "android.permission.INTERNET")
+@UsesLibraries(libraries = "firebase.jar")
+public class FirebaseDB extends AndroidNonvisibleComponent implements Component {
+
+  private static final String LOG_TAG = "Firebase";
+
+  private String firebaseURL;
+  private String projectPath;
+  private Handler androidUIHandler;
+  private final Activity activity;
+  private Firebase myFirebase;
+  private ChildEventListener listener;
+
+  /**
+   * Creates a new Firebase component.
+   *
+   * @param container the Form that this component is contained in.
+   */
+  public FirebaseDB(ComponentContainer container) {
+    super(container.$form());
+    // We use androidUIHandler when we set up operations that run asynchronously 
+    // in a separate thread, but which themselves want to cause actions
+    // back in the UI thread.  They do this by posting those actions
+    // to androidUIHandler.
+    androidUIHandler = new Handler();
+    this.activity = container.$context();
+    Firebase.setAndroidContext(activity);
+    
+    firebaseURL = "https://incandescent-fire-4621.firebaseio.com/";
```

Comment:
This cannot be hardcoded. The default value should error out and indicate the developer (the AI developer) that this property is missing and is required.